From fe5f6cdf4428dcd01f5b060b31eabe44c2f5e468 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:38:58 -0400
Subject: [virt] kvm/x86: expose ADX feature to guest

Message-id: <1406040016-3289-144-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86222
O-Subject: [RHEL7 PATCH v2 143/221] KVM: x86: expose ADX feature to guest
Bugzilla: 1084614 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Liu, Jinsong" <jinsong.liu@intel.com>

ADCX and ADOX instructions perform an unsigned addition with Carry flag and
Overflow flag respectively.

Signed-off-by: Xudong Hao <xudong.hao@intel.com>
Signed-off-by: Liu Jinsong <jinsong.liu@intel.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 49345f13f0830741b94b867cf906c4aad3988306)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/cpuid.c b/arch/x86/kvm/cpuid.c
index bf8cac8..246bc71 100644
--- a/arch/x86/kvm/cpuid.c
+++ b/arch/x86/kvm/cpuid.c
@@ -303,7 +303,8 @@ static inline int __do_cpuid_ent(struct kvm_cpuid_entry2 *entry, u32 function,
  /* cpuid 7.0.ebx */
  const u32 kvm_supported_word9_x86_features =
   F(FSGSBASE) | F(BMI1) | F(HLE) | F(AVX2) | F(SMEP) |
-  F(BMI2) | F(ERMS) | f_invpcid | F(RTM) | F(RDSEED);
+  F(BMI2) | F(ERMS) | f_invpcid | F(RTM) | F(RDSEED) |
+  F(ADX);
 
  /* all calls to cpuid_count() should be made on the same cpu */
  get_cpu();
-- 
1.7.1